From 9457b3169d9c09b4255322c588c59df402fe3196 Mon Sep 17 00:00:00 2001
From: Dmitry Safonov <dima@arista.com>
Date: Thu, 10 May 2018 19:14:47 +0100
Subject: [PATCH 7/8] criu/proc: Define BUF_SIZE without PAGE_SIZE dependency

PAGE_SIZE will be a variable value on platforms where it can be
different due to large pages.
And looks like (c) there is no reason for BUF_SIZE == PAGE_SIZE,
so let's keep it as it was, rather than complicating it with dynamic
allocation for the buffer.

Signed-off-by: Dmitry Safonov <dima@arista.com>
Signed-off-by: Andrei Vagin <avagin@virtuozzo.com>
---
 criu/proc_parse.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/criu/proc_parse.c b/criu/proc_parse.c
index 27719416a..edc56bb99 100644
--- a/criu/proc_parse.c
+++ b/criu/proc_parse.c
@@ -61,16 +61,16 @@
 #define SIGEV_THREAD_ID 4       /* deliver to thread */
 #endif
 
+#define BUF_SIZE	4096	/* Good enough value - can be changed */
+
 struct buffer {
-	char buf[PAGE_SIZE];
+	char buf[BUF_SIZE];
 	char end; /* '\0' */
 };
 
 static struct buffer __buf;
 static char *buf = __buf.buf;
 
-#define BUF_SIZE sizeof(__buf.buf)
-
 /*
  * This is how AIO ring buffers look like in proc
  */
-- 
2.17.0

